.ci/env/tbb.sh

```diff
@@ -0,0 +1,26 @@
+sudo apt-get update
+sudo apt-get install build-essential gcc gfortran cmake -y
+git clone https://github.com/oneapi-src/oneTBB.git
+CoreCount=$(lscpu -p | grep -Ev '^#' | wc -l)
+
+rm -rf __deps/tbb
+pushd oneTBB
```

Comment:
@keeranroth thank you for suggestion. 
Currently makefile expect TBB dependency to reside inside `__deps/tbb/`. Changing name of directory would require making changes in other makefiles as well, which at this point I think is not required. @napetrov any comments.

Reply:
I was referring to the `oneTBB` directory name, which is put into the root directory at the moment. This can probably be onetbb-src or something similar, so that the clone at the top of this file would be:
```
git clone https://github.com/oneapi-src/oneTBB.git onetbb-src
```

and it's probably worth doing a shallow clone so that the switch of the branch isn't necessary

```
git clone --depth 1 --branch v2021.11.0 https://github.com/oneapi-src/oneTBB.git onetbb-src
```

Reply:
i think downloading sources archive is a better option that gitclones here. 

In terms of names - i guess the point is to have build directory properly named as build directory and compiler tbb can be put in to dependencies directory. Point is that it would be easier to debug things with clear naming

Reply:
@napetrov, what if we go with shallow clone as suggested by @keeranroth so that same task can be done in less lines of code as compared to archive download. 
```bash
git clone --depth 1 --branch v2021.11.0 https://github.com/oneapi-src/oneTBB.git onetbb-src
```

Reply:
have nothing against git clone. The only point - it would be good to have single place for versions modification. 

Also there is open question on if env/tbb.sh should be merged with download_tbb script from dev folder

Reply:
The original pull request had both in the same file, which I commented I thought looked weird, in that the download script was also an Arm build recipe. I think that it's better with two different scripts, but happy to go with the majority on that one.

Reply:
2 different scipts is option, but then tbb build script for arm should be named properly. For now it's just tbb.sh that would mean that this is universal script. So might be just rename it it to reflect that it's intended to work on arm only and potentially add platform check there.

Reply:
I also think `env/tbb.sh` should be generic. Only catch is for intel and arm, we know dependency path are `__deps/tbb/lnx/lib/intel64/gcc4.8/` and `__deps/tbb/lnx/lib/arm/gcc4.8/` respectively. How do we take care for other architecture?
Our suggestion for deciding architecture directory -
```bash
arch=$(uname -m)
if [ "${arch}" == "x86_64" ]; then
    arch_dir="intel64"
elif [ "${arch}" == "aarch64" ]; then
    arch_dir="arm"
else
    arch_dir=${arch}
fi
```

Reply:
yes, we can put things in this way.  Although - i thought that arm support would be cross compiled ?  i.e. build would happen on x86 machine for arm target. so uname will not necessary return arm platform

Also i would say we can move download_tbb script logic here for Intel and just use prebuild tbb instead of building it from sources.